sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -229,16 +234,24 @@ private Tree createIfTree(TreeMetaData metaData, KtIfExpression element) {
     return new IfTreeImpl(metaData, condition, thenBranch, elseBranch);
   }
 
-  private Tree createParameter(TreeMetaData metaData, PsiElement element) {
-    String name = ((KtParameter) element).getName();
-    NativeKind kind;
+  private List<Tree> createParameterList(Stream<? extends KtParameter> stream) {
```

Comment:
Actually, this method is not necessary since you check the list content in the parent caller method.
So keeping the `if (element instanceof KtParameter)` in the `convertElementToSlangAST` method and using `list(...)` would yield the same result